### PR TITLE
locking heat map legend defaulting min/max right after adjusting 

### PR DESCRIPTION
**Pull request by uyanga-gb**
# Merge Checklist
## :wrench: Issue(s) fixed:
- [ ] Author referenced issue(s) fixed by this PR:
- [ ] Fixes #[FE-5003]
# Related with 
https://github.com/mapd/mapd-immerse/pull/4891

## :smoking: Smoke Test
- [ ] Works in chrome
- [ ] Works in firefox
- [ ] Works in safari
- [ ] Works in ie edge
- [ ] Works in ie 11

## :ship: Merge
- [ ] author crafted PR's title into release-worthy commit message.

### CODE REVIEW

File: src/chart-addons/stacked-legend.js [hunk 1]
```diff
@@ -153,7 +153,8 @@ export function handleLegendLock({ locked, index = 0 }) {
   let redraw = false
   if (typeof color.scale === "object") { // this if or raster-layer-heatmap-mixin.js
     if (color.legend.locked) {
-      layer.setState(setColorScaleDomain_v2(layer.colorDomain()))
+      const colorDomain = color.scale.domain ? color.scale.domain : layer.colorDomain()
```
Comment (clhenrick) [inline src/chart-addons/stacked-legend.js]: the value of `layer.colorDomain()` could be `null` here, does that matter?

Comment (clhenrick) [reply src/chart-addons/stacked-legend.js]: also is passing `color.scale.domain` okay here? It looks like it's passing an array by reference, might be better to clone the array before passing? maybe it doesn't matter.

Comment (uyanga-gb) [reply src/chart-addons/stacked-legend.js]: either color.scale.domain or layer.colorDomain() expects an value. In initial null cases, domain should be populated with auto scale min/max.

Comment (uyanga-gb) [reply src/chart-addons/stacked-legend.js]: In this case even we pass the array by reference, we not directly modifying the array.

**Pull request merged**